jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2423,14 +2423,19 @@ public boolean equalsIgnoreCase(String string) {
 		int currentLength = lengthInternal();
 		
 		/*[IF Sidecar19-SE]*/
+		byte[] buffer = value;
 		// Check if the String is compressed
 		if (enableCompression && count >= 0) {
-			byte[] buffer = new byte[currentLength];
-			compressedArrayCopy(value, 0, buffer, 0, currentLength);
+			if (buffer.length != currentLength) {
+				buffer = new byte[currentLength];
```

Comment:
Is it worthwhile to write the new array back over the existing `value` array?  Once its been right sized, it would avoid allocations on a second `getBytes()` call.

Reply:
That is an interesting idea - it would save the extra allocates in subsequent getBytes() calls, but when you have shared strings ie multiple strings with the same backing array and different lengths, doing so could increase memory usage. I guess since we are having to allocate anyway one could argue the memory has already increased, the difference would be the increase would be permanent as opposed to transient. I'm not sure on the performance implications one way or the other. @DanHeidinga do you have any thoughts? Regardless this is an improvement since we will no longer copy on every call. Perhaps we can put this in and then try to figure out a study to see if there is benefit from storing it back by surveying some benchmarks or something?

Reply:
We would need some performance runs that measure both throughput and footprint with representative workloads to tell if this makes a difference.  I think it would be a valuable investigation, though it might be better to create a new issue to track this idea.

Reply:
I have opened #375 to track this work.